modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -1192,6 +1254,32 @@ impl WalletClientModule {
     }
 }
 
+/// Polls the federation checking if the activated module consensus version
+/// supports safe deposits, saving the result in the db once it does.
+async fn poll_supports_safe_deposit_version(db: Database, module_api: DynModuleApi) {
+    loop {
+        let mut dbtx = db.begin_transaction().await;
+
+        if dbtx.get_value(&SupportsSafeDepositKey).await.is_some() {
+            break;
+        }
+
+        if let Ok(module_consensus_version) = module_api.module_consensus_version().await {
+            if SAFE_DEPOSIT_MODULE_CONSENSUS_VERSION <= module_consensus_version {
+                dbtx.insert_new_entry(&SupportsSafeDepositKey, &()).await;
+                dbtx.commit_tx().await;
+                break;
+            }
+        }
+
+        if is_running_in_test_env() {
+            sleep(Duration::from_secs(1)).await;
+        } else {
+            sleep(Duration::from_secs(3600)).await;
+        }
+    }
```

Comment:
Is keeping a db tx open unnecessarily long a performance problem in any way? For my (naive) `webkv` impl it would be since it couldn't GC deltas from committed txns that are younger than this dbtx.

Reply:
Great question, there's no reason to keep the `dbtx` open for one hour.